prime-router/src/test/kotlin/azure/batch/BatchDeciderTests.kt

```
@@ -75,115 +76,120 @@ class BatchDeciderTests {
             .queueAccess(queueMock)
             .build()
 
-    @BeforeTest
+    @BeforeEach
     fun reset() {
         clearAllMocks()
     }
 
     @Test
-    fun `Test with no receiver getting empty batch file`() {
+    fun `Test receivers with no WhenEmpty action set to will not queue an empty batch`() {
```

These tests all revolved around the functionality of sending empty batches. The actual logic for this was moved out to the method `determineQueueMessageCount` so I updated them to focus on only testing that directly. 

In part this was because the try/catch block on the `run` method swallows any errors. The empty batch tests were silently hitting exceptions in `run`. They were still passing because their assertions were on the results of `determineQueueMessageCount`.

The scenarios have been renamed to more clearly convey what they are testing:
- When a receiver does not define WhenEmpty, it should default to not sending an empty batch.
- When a receiver has SEND set as the WhenEmpty action, and it should send empty batches as often as the batch timer is for that receiver.
- When a receiver has SEND set as the WhenEmpty action, and it is set to only send one per day.